43_tpm_1_2_support.md

```diff
@@ -0,0 +1,176 @@
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+
+- [ ] Enhancement issue in release milestone, which links to pull request in [keylime/enhancements]
+- [ ] Core members have approved the issue with the label `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place
+- [ ] User-facing documentation has been created in [keylime/keylime-docs]
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+## Summary
+
+This enhancement proposes to extend support for TPM 1.2 which was removed in Keylime v6.0.0
+
+
+## Motivation
+
+Modern servers usually have TPM 2.0 chip rather than TPM 1.2, however not all vendors support 
+measured boot with TPM 2.0 in legacy boot mode which means that the latest Keylime software 
+cannot be used for remote attestation of those legacy systems. 
+Extended support for TPM 1.2 will allow Keylime users to use the latest Keylime versions 
+without backporting patches for potential vulnerabilities that may arise in the future. 
+With the enhancement the same Keylime baseline can be used for remote attestation of both
+TPM 1.2 and TPM 2.0 systems which will make the future migration process from TPM 1.2 to TPM 2.0
+more smooth.
+
+
+### Goals
+
+Add TPM 1.2 support on master branch.
+
+### Non-Goals
+
+<!--
+What is out of scope for this enhancement?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+Extend support for TPM 1.2.
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this enhancement is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+enhancement ecosystem.
+
+How will security be reviewed and by whom?
+-->
+While TPM 1.2 uses weaker algorithms (e.g. SHA1), Keylime security (components, deployments and operations) remains the same  regardless of using TPM 1.2 or TPM 2.0.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
```

Comment:
This part really needs to be fleshed out. One of my main concerns with adding this back in is ongoing testing and support. Can we do testing with a software TPM for 1.2 that is still maintained in containers like we do today? Will we need parallel test suite runs in CI for 1.2 vs 2.0? Lots of details on the testing side to flesh out.